Client/Frontend/Browser/HomePanel/NTPDownloader.swift

```diff
@@ -0,0 +1,378 @@
+        guard let supportDirectory = FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask).first else {
+            return nil
+        }
+        
+        let downloadsURL = supportDirectory.appendingPathComponent(NTPDownloader.ntpDownloadsFolder)
+        let metaDataURL = downloadsURL.appendingPathComponent(NTPDownloader.metadataFile)
+        
+        if !FileManager.default.fileExists(atPath: metaDataURL.path) {
+            return nil
+        }
+        
+        do {
+            let metadata = try Data(contentsOf: metaDataURL)
+            let itemInfo = try JSONDecoder().decode(NTPItemInfo.self, from: metadata)
+            
+            let logo = NTPLogo(imageUrl: metaDataURL.appendingPathComponent(itemInfo.logo.imageUrl).path,
+                               alt: itemInfo.logo.alt,
+                               companyName: itemInfo.logo.companyName,
+                               destinationUrl: itemInfo.logo.destinationUrl)
+            
+            let wallpapers = itemInfo.wallpapers.map({
+                return NTPWallpaper(imageUrl: metaDataURL.appendingPathComponent($0.imageUrl).path,
+                                    focalPoint: $0.focalPoint)
+            })
+            
+            return NTPItemInfo(logo: logo, wallpapers: wallpapers)
+        } catch {
+            logger.error(error)
+        }
+        
+        return nil
+    }
+    
+    private func getETag() -> String? {
+        guard let supportDirectory = FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask).first else {
+            return nil
+        }
+        
+        let downloadsURL = supportDirectory.appendingPathComponent(NTPDownloader.ntpDownloadsFolder)
+        if !FileManager.default.fileExists(atPath: downloadsURL.path) {
+            return nil
+        }
+        
+        let etagURL = downloadsURL.appendingPathComponent(NTPDownloader.etagFile)
+        if !FileManager.default.fileExists(atPath: etagURL.path) {
+            return nil
+        }
+        
+        return try? String(contentsOfFile: etagURL.path, encoding: .utf8)
+    }
+    
+    private func setETag(_ etag: String) {
+        guard let supportDirectory = FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask).first else {
+            logger.error("Cannot find Support Directory for writing NTPDownloader ETag")
+            return
+        }
+        
+        let downloadsURL = supportDirectory.appendingPathComponent(NTPDownloader.ntpDownloadsFolder)
+        
+        do {
+            try FileManager.default.createDirectory(at: downloadsURL, withIntermediateDirectories: true, attributes: nil)
+        } catch {
+            logger.error("NTPDownloader: \(error)")
+            return
+        }
+        
+        let etagURL = downloadsURL.appendingPathComponent(NTPDownloader.etagFile)
+        
+        do {
+            let etag = etag.replacingOccurrences(of: "\"", with: "")
```

Comment:
Why drop the backward slash here?
Will this work when sending etag to the server for comparison?